clueless/state.py

```diff
@@ -41,33 +51,63 @@ def sock_addr(self) -> str:
+        return self.state is UserState.REGISTERED
 
     async def dispatch_message(self, message):
+        print("Dispatch_message function:")
         if isinstance(message, CreateGame):
-            self.logger.debug(f"Received crate game")
-            game = GameState()
-            id = game.id
-            self.logger.debug(f'Created game {id}')
-            GAMES[id] = game
-            await GAMES[id].add_user(self)
+            if self.game is None:                       # If this player instance is not in a game
+                self.game = GameState()                 # Create an instance of GameState class, associate it with Player instance
+                id = self.game.id                            
+                self.logger.debug(f'Created game {id}')
+                GAMES[id] = self.game                   # Place in a dictionary of games
+                await GAMES[id].add_user(self)          # Add user into recently created class
+            else:
+                self.send_message(Status("Player is already in a game")) # RuntimeWarning: coroutine 'Player.send_message' was never awaited
+        
         elif isinstance(message, JoinGame):
-            self.logger.debug(f'Joining game {message.id}')
-            await GAMES[message.id].add_user(self)
+            print("JoinGame instance")
+            if self.game is None:                           # Check whether this player instance is assocaited with a game
+
+                self.logger.debug(f'Player Validated')      
+                self.game = GAMES[message.id]               # Instance of player gets its own game association
+
+                if self.game.id in GAMES:                   # Check whether the game.id exists in dictionary
+                    self.logger.debug(f'Game {message.id} Validated')
+                    await GAMES[message.id].add_user(self)
+                else:
+                    self.logger.debug(f'Game ID {message.id} DOES NOT exist') # RuntimeWarning: coroutine 'Player.send_message' was never awaited
+                    self.send_message(Status("Game ID does NOT exist"))
+            else:
+                self.logger.debug(f'Player is in a game')
+                self.send_message(Status("Player is already in a game"))
+                
         elif isinstance(message, Register):
-            self.logger.debug(f'Registering as {message.display_name} with {message.character}')
-            self.character = message.character
-            self.display_name = message.display_name
-            await self.game.register_user(self, message)
+            print("Register instance")
+            if self.game.id in GAMES and not self.is_registered:    # Not quite sure if this is enough validation to say a player is in a certain game
+                self.logger.debug(f'Registering as {message.display_name} with {message.character}')
+                self.character = message.character                  # Setting the instance character
```

Comment:
We may actually want to move so that `GameState` sets the value in `register_user` after validation. If the player registers as a character that's already taken, this will still set it. Either it'd have to be moved to `register_user`, or `register_user` would have to unset it if the registration isn't valid. Something like:
```
if self.characters[msg.character] is not None:
  player.character = None
  await player.send_message(some error message)
```